src/main/kotlin/com/exactpro/th2/check1/metrics/BufferMetric.kt

```diff
@@ -15,74 +15,41 @@ package com.exactpro.th2.check1.metrics
 import com.exactpro.th2.common.metrics.DEFAULT_SESSION_ALIAS_LABEL_NAME
 import io.prometheus.client.Gauge
-import mu.KotlinLogging
-import org.openjdk.jol.info.GraphLayout
-import java.util.Queue
-import java.util.LinkedList
 import java.util.concurrent.ConcurrentHashMap
 
 object BufferMetric {
-    private val logger = KotlinLogging.logger {}
 
     private val actualBufferCountMetric: Gauge = Gauge
             .build("th2_check1_actual_cache_number", "The actual number of messages in caches")
             .labelNames(DEFAULT_SESSION_ALIAS_LABEL_NAME, DEFAULT_DIRECTION_LABEL_NAME)
             .register()
-    private val actualBufferSizeMetric: Gauge = Gauge
-            .build("th2_check1_actual_cache_size", "The actual size of messages in caches")
-            .labelNames(DEFAULT_SESSION_ALIAS_LABEL_NAME, DEFAULT_DIRECTION_LABEL_NAME)
-            .register()
 
-    private val bufferMessagesSizeBySessionKey: MutableMap<SessionKey, Queue<Long>> = ConcurrentHashMap()
+    private val bufferMessagesSizeBySessionKey: MutableMap<SessionKey, Int> = ConcurrentHashMap()
     private var maxBufferSize: Int = -1
 
     fun configure(configuration: Check1Configuration) {
         this.maxBufferSize = configuration.messageCacheSize
     }
 
-    fun processMessage(sessionKey: SessionKey, message: Message) {
-        val calculatedMessageSize = calculateMessageSize(message)
+    fun processMessage(sessionKey: SessionKey) {
         val labels = arrayOf(sessionKey.sessionAlias, sessionKey.direction.name)
 
-        bufferMessagesSizeBySessionKey.compute(sessionKey) { _, bufferedMessageSize ->
-            if (bufferedMessageSize == null) {
-                incrementStats(calculatedMessageSize, labels)
-                return@compute LinkedList<Long>().apply { add(calculatedMessageSize) }
+        bufferMessagesSizeBySessionKey.compute(sessionKey) { _, bufferedMessagesNumber ->
+            if (bufferedMessagesNumber == null) {
+                actualBufferCountMetric.labels(*labels).inc()
+                return@compute 1
             }
 
-            if (bufferedMessageSize.size >= maxBufferSize) {
-                decrementStats(bufferedMessageSize.poll(), labels)
+            if (bufferedMessagesNumber == maxBufferSize) {
+                return@compute bufferedMessagesNumber
             }
 
-            bufferedMessageSize.add(calculatedMessageSize)
-            incrementStats(calculatedMessageSize, labels)
-
-            return@compute bufferedMessageSize
-        }
-    }
+            actualBufferCountMetric.labels(*labels).inc()
 
-    private fun calculateMessageSize(message: Message): Long {
-        val parsedInstance = GraphLayout.parseInstance(message)
-        if (logger.isTraceEnabled) {
-            logger.trace("Foot print for message with id: '{}'\n{}", message.metadata.id.toJson(), parsedInstance.toFootprint())
+            return@compute bufferedMessagesNumber.inc()
         }
```

Comment:
Looks to me that we can replace all this with the this function. It's a bit clearer in my opinion.
```kotlin
bufferMessagesSizeBySessionKey.compute(sessionKey) { _, old ->
  min(maxBufferSize, (old ?: 0) + 1).also {
    if (it != old) {
       actualBufferCountMetric.labels(*labels).inc()
    }
  }
}
```